File: backend/modules/task_management/repository/task_repository.go
package repository

import (
	taskManagementContract "stm/modules/task_management/domain/contract"
	taskDto "stm/modules/task_management/domain/dto"
	taskManagementEntity "stm/modules/task_management/domain/entity"
	sharedModel "stm/shared/model"
	gormUtil "stm/shared/utility/gorm"

	"gorm.io/gorm"
)

type taskRepo struct {
	db *gorm.DB
}

func (l *taskRepo) Save(task *taskManagementEntity.Task) (taskManagementEntity.Task, error) {

	err := l.db.Where(taskManagementEntity.Task{Id: task.Id}).
		Assign(task).
		FirstOrCreate(&taskManagementEntity.Task{}).Error

	return *task, err
}

func (l *taskRepo) GetAllTasks(filter *sharedModel.CriteriaFilter) ([]taskDto.TaskRecord, int64, error) {
	tasks := []taskDto.TaskRecord{}

	query := l.db.Table("tasks t").
		Joins("JOIN task_status ts ON t.status_id = ts.id").
		Joins("JOIN users u ON t.created_by_id = u.id").
		Select(`
		t.*, ts.name as status, 
		ts.color, 
		CONCAT(u.name, ' ', u.last_name) as created_by
	`)

	if search, ok := (filter.Filters)["search"]; ok && search != "" {
		query = query.Where("CONCAT(t.title, ' ', t.description) LIKE ?", "%"+search.(string)+"%")
	}

	if status, ok := (filter.Filters)["status"]; ok && status != "" {
		query = query.Where("t.status_id = ?", status)
	}

	total := gormUtil.GetCount(l.db, query)

	query = query.Limit(int(filter.Limit))
	query = query.Offset(int(filter.Skip))

	query = query.Order("t.id desc")
	l.db.Raw("?", query).Scan(&tasks)

	return tasks, total, nil
}

func (l *taskRepo) GetTaskById(taskId uint) (taskManagementEntity.Task, error) {
	task := taskManagementEntity.Task{}

	err := l.db.Where(&taskManagementEntity.Task{
		Id: taskId,
	}).Find(&task).Error

	return task, err
}

func (l *taskRepo) ChangeTaskStatus(taskId uint, status uint) error {
	task := taskManagementEntity.Task{}

	err := l.db.Model(&task).
		Where("id = ?", taskId).
		Update("status_id", status).Error

	return err
}

func (l *taskRepo) GetAllTasksStatus() ([]taskManagementEntity.TaskStatus, error) {
	taskStatus := []taskManagementEntity.TaskStatus{}
	err := l.db.Find(&taskStatus).Error
	return taskStatus, err
}

func NewTaskRepo(db *gorm.DB) taskManagementContract.TaskRepository {
	return &taskRepo{
		db: db,
	}
}

File: frontend/src/modules/task_management/repository/taskRepository.ts
import type { TaskContract } from '../domain/contract/taskContract'
import type { Task, TaskFilter, TaskRecord, TaskStatus } from '../domain/entity/task'
import { post, get, patch, DEFAULT_API_PATH } from '@/modules/shared/http_handler'
import { getTaskRecords, getTaskSingleRecords } from './adapters/taskRecords'

const loadTasks = async (filter: TaskFilter | null): Promise<GridRecord> => {
	const queryFilter = {
		limit: filter?.limit,
		skip: filter?.skip,
		search: filter?.filters.search,
		createdBy: (filter?.filters?.createdBy === undefined || filter?.filters?.createdBy === null)  
				? null : filter.filters.createdBy,
		assignedTo: (filter?.filters?.assignedTo === undefined || filter?.filters?.assignedTo === null)  
				? null : filter.filters.assignedTo,                             
		status: (filter?.filters?.statusId === undefined || filter?.filters?.statusId === null)  
				? null : filter.filters.statusId
	}

	const response = await get(DEFAULT_API_PATH + '/tasks_management/tasks', queryFilter );
	return getTaskRecords(response);
}

const saveTask = async (task: Task): Promise<TaskRecord | null> => {
	const requestData = await post(DEFAULT_API_PATH + '/tasks_management/tasks', task);
	return getTaskSingleRecords(requestData);
}

const getTaskById = async (taskId: number): Promise<Task> => {
	const requestData = await get(DEFAULT_API_PATH + '/tasks_management/tasks/' + taskId);
	return requestData.data;
}

const changeTaskStatus = async (taskId: number, status: number): Promise<boolean> => {
	const requestData = await patch(DEFAULT_API_PATH + `/tasks_management/tasks/${taskId}/${status ? 1 : 0}`, {} );
	return requestData.status === 200;
}

const taskExists = async (taskName: string): Promise<boolean> => {
	const requestData = await get(DEFAULT_API_PATH + `/tasks_management/tasks/${taskName}/exists`, {} );
	return requestData.data.exists;
}

const getTaskStatus = async (): Promise<TaskStatus[]> => {
	const requestData = await get(DEFAULT_API_PATH + `/tasks_management/task_status`, {} );
	return requestData.data;
}

export default {
	loadTasks,
	saveTask,
	getTaskById,
	changeTaskStatus,
	taskExists,
	getTaskStatus
} as TaskContract